internal/setup/setup.go

```diff
@@ -623,3 +650,75 @@ func Select(release *Release, slices []SliceKey) (*Selection, error) {
 
 	return selection, nil
 }
+
+// convertPathInfoToYamlPath converts a PathInfo object to a yamlPath object.
+func convertPathInfoToYamlPath(pi *PathInfo) (*yamlPath, error) {
+	if pi == nil {
+		return nil, nil
+	}
+	path := &yamlPath{
+		Mode:    yamlMode(pi.Mode),
+		Mutable: pi.Mutable,
+		Until:   pi.Until,
+		Arch:    yamlArch{List: pi.Arch},
+	}
+	switch pi.Kind {
+	case DirPath:
+		path.Dir = true
+	case CopyPath:
+		path.Copy = pi.Info
+	case TextPath:
+		path.Text = &pi.Info
+	case SymlinkPath:
+		path.Symlink = pi.Info
+	case GlobPath:
+	default:
+		return nil, fmt.Errorf("unrecognised PathInfo type: %s", pi.Kind)
+	}
+	return path, nil
+}
+
+// convertSliceToYamlSlice converts a Slice object to a yamlSlice object.
+func convertSliceToYamlSlice(s *Slice) (*yamlSlice, error) {
+	if s == nil {
+		return nil, nil
+	}
+	slice := &yamlSlice{
+		Essential: make([]string, len(s.Essential)),
+		Contents:  make(map[string]*yamlPath, len(s.Contents)),
+		Mutate:    s.Scripts.Mutate,
+	}
+	for idx, sliceKey := range s.Essential {
+		slice.Essential[idx] = sliceKey.String()
+	}
+	for contPath, pathInfo := range s.Contents {
```

Comment:
`contPath` reads funny to me. `contentPath` would be better. But what about just using `path` for this one, which is a real `PathInfo`, and `yamlPath` for the variable name below?

Reply:
Changed in https://github.com/canonical/chisel/pull/101/commits/b886f473b8b48064e4138ed6022bf165721e43be. Thanks.